### DISCUSSION TITLE
Using MUI X Pro in an open-source project?

### DISCUSSION THREAD

**Discussion by openscript** [mui/mui-x#1010] [Archive]
Hello everybody

Is it possible to use MUI X Pro in an opensource project? Buying one license or so is fine for us, but more would blow up the budget.

Thanks :)

**Comment by mbrookes** [comment]
Hi @openscript, yes you can use MUI X Pro/MUI X Premium in an open-source project. However, there are two important aspects to pay attention to.

- First, the number of licenses. It must match with the maximum number of concurrent developers working on the project on any given day. If you only have a few contributions, 2 licenses are likely enough, as the seats aren't named. Otherwise, it would need to grow up to the cap limit we set.
- Second, the license. It sets legal constraints on what people can do with the software. There are two different use cases:
  - The nominal use case. You make the license key private to your organization. You can deploy and use the software as part of a SaaS product or for an internal use case. People forking the software outside of your organization need to purchase new licenses to cover their use case.
  - Advanced use case, sublicensing. The [EULA](https://mui.com/legal/mui-x-eula/) has a "Deployment" article that covers how MUI X Pro and Premium can be sublicensed. In practice, it restricts the open-source license that you have put in place.

Note that DataGridPro without a license key is still fully functional (it shows a watermark and a warning in the console). Our EULA allows it to be used for testing purposes (to be clear, no development purposes but tests and experimentations).

**Reply by openscript** [reply]
Thank you for answering so quickly. So it will show a watermark during development only? Will it work inside a PWA (offline) for the users without showing a watermark?

XGrid won't be the only Premium material component right? I've seen that other components like the Tree Selection also needs to be licensed.

**Reply by oliviertassinari** [reply]
Correct and the other commercial components will have the same license.

**Reply by oliviertassinari** [reply]
> So it will show a watermark during development only? Will it work inside a PWA (offline) for the users without showing a watermark?

Yes, once you license the right number of developers and install the license key (it works offline).

**Reply by openscript** [reply]
@oliviertassinari an open source project with a community won't work like that. If you want to contribute, you need to buy a license for 249$ per annum first. It's like asking for money from the receiver of a birthday present.

Anyway, I understand that it's a huge effort to develop components like a grid. I would be happy to pay something (from my own pocket) to support that, if the license somehow integrates with the project.

**Reply by oliviertassinari** [reply]
If you are willing to make your open source project under the AGPL license, https://github.com/mui-org/material-ui-x/discussions/1010#discussioncomment-347087 could potentially work in the long term.

Do you have more context on this open source project? It would help.

Also mind that this repository is about the advanced components (MIT or not). The aim is to move the Autocomplete, TreeView, Pickers here (without changing the license).

**Reply by openscript** [reply]
It's about different projects. The most recent is: https://github.com/timelet/timelet

**Reply by gtanchak** [reply]
@oliviertassinari @mbrookes
If our organisation used Pro components and our 3 developer working on that component and they are part of our organisation, So how many license key we need to buy?

<img width="407" alt="Screenshot 2023-10-14 at 5 35 48 PM" src="https://github.com/mui/mui-x/assets/81280194/037cbcf5-2cf7-4017-98d2-19ebc1790cf6">

**Reply by oliviertassinari** [reply]
@gtanchak your organization needs to license all the developers loading the pro components, either raw or wrapped. The license is structured in a way that makes backend engineers not needing a license. Usually all the full-stack and frontend engineers working on the project needs a license.

**Reply by gtanchak** [reply]
@oliviertassinari Thanks!!

But, currently if i have buy one license key and used in one repository and in that repository we have 3 developer's so should I need to but other license ? Or it will work?

**Reply by oliviertassinari** [reply]
If these 3 developers are loading the page in dev mode with the Pro components, the project needs 3 licenses.

To be clear a license key is meant as a reminder, it's unique configured once in the code.
But what matters from a IP perspective are the licenses (not the license key).

**Reply by gtanchak** [reply]
Thanks for the clarification.

**Reply by HyungwonJang0327** [reply]
@mbrookes @oliviertassinari 

Hello, I would like to use mui datagrid pro in 3 private repositories.

I'm the only worker at the moment, and one who previously ran the project together left the team last week.

How many licenses do I need to buy,
I guess one licenses, is right?

**Comment by oliviertassinari** [comment]
We have been considering a dual license with [AGPL](https://www.quora.com/What-is-the-difference-between-GPL-AGPL-and-LGPL-licenses). It could become relevant at some point. At this point, we are waiting for a large open source project to ask about it. It could be interesting for us to broaden reach.

**Comment by epacke** [comment]
Hello
I'm in the same situation here. Working on an open source project and I've been thinking to replace the frontend with React + MUI.

If I understood the above discussion correctly:

- Mui has no Open Source program for Pro/Premium?
- If I buy Pro from my own pocket I'd have to pay an additional license for each dev contributing?

The project I wish to use MUI for is:

https://github.com/net-utilities/BigIPReport

It's under GPL 3.0 license.

Kind regards,
Patrik

**Reply by oliviertassinari** [reply]
@epacke the previous discussions on this thread are up to date, they should cover your questions. 

Contributors aren't named, so as long as you ou have enough licenses to cover concurrent contributions, you should be good.

For the dual license option, we couldn't envision a clear benefit of having it so we didn't make the change.

**Reply by bluefire2121** [reply]
What about for non-profits like scholastic institutions? Is there a discount? Likewise is there a discount if all we need is a single component such as the date range selector?

**Comment by oliviertassinari** [comment]
> What about for non-profits like scholastic institutions? Is there a discount? Likewise is there a discount if all we need is a single component such as the date range selector?

@bluefire2121 If this is meant for educational purposes to students, we can think of a special license. You can contact us a sales@mui.com.

> Likewise is there a discount if all we need is a single component such as the date range selector?

There isn't, the Pro plan might grow by limiting the access to 2-3 non MIT components features to stay compatible with its low pricing point.

**Comment by oliviertassinari** [comment]
> Hello, I would like to use mui datagrid pro in 3 private repositories.
I'm the only worker at the moment, and one who previously ran the project together left the team last week.
How many licenses do I need to buy, I guess one license, is right?

@HyungwonJang0327 Correct, one
